CLI is complete, Tests are working.

swsnider opened the pull request:
GUI is a stub -- pay no attention to it :)

swsnider committed:
Getting rid of extraneous binary file

swsnider committed:
Renamed SQLToYAML Assembly

There was an ugly name clash between the assembly name for the class
library and the class I'm developing, which caused more typing, and was
just ugly. Therefore, I've refactored it to something more pleasing.

swsnider committed:
Program now seems to work with hardcoded config.

Next Step: Unit Tests!

swsnider committed:
I'm fairly confident this produces correct YAML

swsnider committed:
Added third-party command line option parser

swsnider committed:
CLI is confirmed working, with awesome flags.

swsnider committed:
Refactored SQL Server access

In the next couple of commits, when I implement actual unit tests, it'll
be useful to have a dependency-injected data source for mocking.

swsnider committed:
YAMLConversion now has tests

swsnider committed:
SQLToYAML is now tested.

ayust commented on EVE-Val SQL to YAML Tool/SQL To YAML CLI/Options.cs:
```diff
@@ -0,0 +1,39 @@
+        [Option(null, "connection_string", DefaultValue="Data Source=localhost; Initial Catalog=ebs_DATADUMP; Integrated Security=SSPI;", HelpText = "The database connection string to use.")]
+        public string connectionString { get; set; }
+
+        [Option("t", "table", DefaultValue="*", HelpText="The database table to extract. Defaults to '*'.")]
```
The default being a wildcard implies that (a) wildcards can be used and (b) multiple tables can be extracted at the same time. If this is the case, `table(s)` would probably be more appropriate, and a mention of what wildcards are allowed.

ayust commented on EVE-Val SQL to YAML Tool/SQL To YAML CLI/Options.cs:
```diff
@@ -0,0 +1,39 @@
+{
+    class Options
+    {
+        [Option(null, "connection_string", DefaultValue="Data Source=localhost; Initial Catalog=ebs_DATADUMP; Integrated Security=SSPI;", HelpText = "The database connection string to use.")]
```
This is really long. Does C# allow this to be linewrapped?

(In general, it's best to keep code width below the ~120 characters visible in GitHub diff views.)

swsnider committed:
Complying with code review

Note: I decided not to split the per-row logic into its own method
because I need to fire the progressMade events, and that's awkward to
split across the function boundary.

swsnider commented:
PTAL.

ayust commented:
LGTM
